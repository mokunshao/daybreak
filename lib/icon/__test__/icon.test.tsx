import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Icon from '../icon';

describe('Icon', () => {
  it('是个 svg', () => {
    const json = renderer.create(<Icon name="qq" />).toJSON();
    expect(json).toMatchSnapshot();
  });
  it('onClick', () => {
    const fn = jest.fn();
    const icon = mount(<Icon name="qq" onClick={fn} />);
    icon.find('svg').simulate('click');
    expect(fn).toBeCalled();
  });
});
