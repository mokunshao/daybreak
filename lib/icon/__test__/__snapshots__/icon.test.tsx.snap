// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon 是个 svg 1`] = `
<svg
  className="kaka-icon"
>
  <use
    xlinkHref="#qq"
  />
</svg>
`;
